PXDConverter/Program.cs

```diff
@@ -28,132 +46,549 @@ private static void Main(string[] args)
+            if (s[s.Length - 1].EndsWith(".pxd"))
+            {
+                return new PXDHeader(s[s.Length - 1][..^4], s.ElementAtOrDefault(s.Length - 2) ?? string.Empty, string.Empty, samples, isWave);
+            }
+            else if (s.Length < 3)
+            {
+                return new PXDHeader(s[s.Length - 1], Path.GetFileName(Path.GetDirectoryName(Path.GetDirectoryName(filepath))!) + " DEMO", Path.GetFileName(Path.GetDirectoryName(filepath)!), samples, isWave);
+            }
+            else
+            {
+                return new PXDHeader(s[0] + (s.ElementAtOrDefault(1) ?? string.Empty), $"{s.ElementAtOrDefault(2) ?? string.Empty} {s.ElementAtOrDefault(3) ?? string.Empty}".Trim(), s.ElementAtOrDefault(4) ?? string.Empty, samples, isWave);
+            }
+        }
 
-            if (string.IsNullOrEmpty(argPath))
-                argPath = Directory.GetCurrentDirectory();
-            
-            string[] filePaths = Directory.GetFiles(argPath, "*.pxd", SearchOption.AllDirectories);
+        private static (int, R) processBytes<R>(byte[] arr, int p, int size, Func<byte[], R> convert)
+        {
+            R res = convert(arr[p..(p + size)]);
+            return (p + size, res);
+        }
+
+
+        private static (int, byte[]) readBytes(byte[] arr, int p, int size)
+        {
+            return (p + size, arr[p..(p + size)]);
+        }
 
-            if (filePaths.Length == 0)
+        private static bool isLastRecord(byte[] descriptorBytes, int p)
+        {
+            p += 8;
+            (p, short sampleRateByte) = processBytes(descriptorBytes, p, 2, b => BitConverter.ToInt16(b));
+            return sampleRateByte == 0;
+        }
+
+        // sampleRate or maybe samplesCount? not sure
+        private record MultiPXDRecord(int id, int group, string name, int offset, int lenght, int sampleRateByte, int type);
+
+        private static byte[] ReadFileFromTo(string fileName, int offset, int lenght)
+        {
+            byte[] fileData;
+            using (FileStream fs = File.OpenRead(fileName))
             {
-                Console.WriteLine("Couldn't find any .pxd file!");
-                return;
+                fs.Position = offset;
+                using (BinaryReader binaryReader = new BinaryReader(fs))
+                {
+                    fileData = binaryReader.ReadBytes(lenght);
+                }
             }
-            
-            foreach (var path in filePaths)
+            return fileData;
+        }
+
+        private static (int, MultiPXDRecord) readBinaryPXDRecord(byte[] descriptorBytes, int p)
+        {
+            (p, byte[] fileSeparator) = readBytes(descriptorBytes, p, 2);
+            (p, byte id) = processBytes(descriptorBytes, p, 1, b => b[0]);
+            (p, byte group) = processBytes(descriptorBytes, p, 1, b => b[0]);
+            p += 2; // serarator
+            (p, short nameLenght) = processBytes(descriptorBytes, p, 2, b => BitConverter.ToInt16(b));
```

Comment:
If I'm not wrong, every lambda like this one can be shortened to method group

Reply:
just `BitConverter.ToInt16` doesn't work here